### ISSUE TITLE
Request - is there a way to look at the sources for taxa used by a collection?

### ISSUE THREAD

**Issue by happiah-madson** [ArctosDB/arctos#8166]
Help us understand your request (check below):
- [x] search or data request (help with SQL or provide the specific fields you are trying to retrieve for your collection)
- [ ] add a new Github user to the Arctos Users team
- [ ] a new barcode series
- [ ] a new term for data entry or management 
- [ ] a change or new report or label
- [ ] update to an Agent (split, merge, or other type of change)
- [ ] bulkload changes (to Agents, Identifiers, etc that are not already available as a bulkloading tool)
- [ ] new feature or otherwise a good idea to make Arctos even better
- [ ] quarantine taxon name and suggested replacement (please supply links)
- [ ] other

**Describe what you're trying to do** 
I'm looking at the summary information for taxonomy in the OGL collection and realized that some of our records are linked to taxon records that have sources that aren't WoRMS via Arctos. In some cases this is logical (the taxon doesn't exist in WoRMS) but in other cases that taxon exists in WoRMS but the WoRMS data hasn't been linked to Arctos, and so our source is Arctos Plants (e.g., https://arctos.database.museum/name/Valonia%20fastigiata#ArctosPlants). Whenever possible, we'd like our taxonomy source to be WoRMS, so I am wondering if there is a way for me to find the set of taxa in the OGL collection whose source is not WoRMS via Arctos.

**Comment by dustymc** [comment]
[temp_oglnowva.csv.zip](https://github.com/user-attachments/files/17264364/temp_oglnowva.csv.zip)

(Also: I don't believe your 'unidentifiable's - I doubt you've got many tubes of https://arctos.database.museum/name/motorcycle or https://arctos.database.museum/name/Nickel - maybe https://arctos.database.museum/name/life?)

**Comment by happiah-madson** [comment]
Fully agree w/ the unidentifiable's....we just didn't use to have a way to handle it! Thank youuuuuuuu

**Comment by happiah-madson** [comment]
Okay, I'm finally getting around to looking at this. I see that you've given me a list of species names and the count of that species name in the OGL collection. Is there a logical systematic way that I can update the source? The steps to me seem like: 1) manually create all of these taxa in WoRMS via Arctos following @sharpphyl's protocol. Manually look at the records for each species name and er, somehow reassign the source in the identification.

Is there a way to do this that I don't know about? Or is there a template that I could fill out for @dustymc that you could use and do magic under the covers? thanks!

**Comment by dustymc** [comment]
>given me a list 

yup

>species

nope, identifications

> logical systematic way that I can update the source?

All my ideas are still https://handbook.arctosdb.org/documentation/taxonomy.html (eg manage what you care about). Not sure how that lines up with worms...

> manually create all of these taxa 

In case it adds any clarity: the taxa exist, you need the classifications.

>somehow reassign the source

That's magic, if the source is there Arctos will find and use it, no need to do anything.

>template

Don't think it's needed, but here's my query in case that's handy (it should-probably-maybe run in writesql).

```sql
select scientific_name, count(*) c from flat where guid_prefix='OGL:Genomic' and previousidentifications::varchar not like '%WoRMS (via Arctos)%' group by scientific_name order by scientific_name;
```

**Comment by happiah-madson** [comment]
> In case it adds any clarity: the taxa exist, you need the classifications.

absolutely does, thank you.

**Comment by happiah-madson** [comment]
Quick follow up: I've done a bunch of classification updating (in December) and just re-ran the SQL you provided above. It worked beautifully, but some records still seem to be pulling non-WoRMS classifications. Is this because Arctos is just chugging along doing its thing and it needs more time?

**Comment by dustymc** [comment]
>chugging

Probably, or it's stuck (that's a wildly complicated system, something falling through the cracks isn't too surprising- but still please let me know if you can re-create and I'll plug that hole).

You can hit the throttle. Change my SQL a bit...

```
select scientific_name, guid from flat where guid_prefix='OGL:Genomic' and previousidentifications::varchar not like '%WoRMS (via Arctos)%' order by scientific_name

```

feed the GUIDs to the listerizer, feed that to search, then tools/request refresh.

Except that's mostly "maybe a motorcycle?" IDs, so....

```

 Barabattoai amicorum cf. | OGL:Genomic:4659
 Caulerpa nummularia      | OGL:Genomic:2674
 Caulerpa racemosa        | OGL:Genomic:32
 Chondrus crispus         | OGL:Genomic:40709
 Chondrus crispus         | OGL:Genomic:40710
 Chondrus crispus         | OGL:Genomic:40697
 Chondrus crispus         | OGL:Genomic:40327
 Chondrus crispus         | OGL:Genomic:40696
 Crepidula chilenis       | OGL:Genomic:7512
 Crepidula chilenis       | OGL:Genomic:7508
 Crepidula chilenis       | OGL:Genomic:7450
 Crepidula chilenis       | OGL:Genomic:7514
 Crepidula chilenis       | OGL:Genomic:7513
 Epactionotus aky         | OGL:Genomic:16308
 Holothuria atra cf.      | OGL:Genomic:3803
 Notoacmea persona        | OGL:Genomic:23452
 Ophiura figurata cf.     | OGL:Genomic:12737
 Ophiura figurata cf.     | OGL:Genomic:12738
 Ophiura integra          | OGL:Genomic:12810
 Ophiura integra          | OGL:Genomic:12807
 Ophiura integra          | OGL:Genomic:12806
 Ophiura integra          | OGL:Genomic:12812
 Ophiura integra          | OGL:Genomic:12811
 Ophiura integra          | OGL:Genomic:12808
 Ophiura integra          | OGL:Genomic:12809
 Ophiura integra cf.      | OGL:Genomic:12740
 Ophiura integra cf.      | OGL:Genomic:12739
 Ophiura integra cf.      | OGL:Genomic:12743
 Ophiura integra cf.      | OGL:Genomic:12744
 Ophiura integra cf.      | OGL:Genomic:12742
 Ophiura integra cf.      | OGL:Genomic:12745
 Ophiura integra cf.      | OGL:Genomic:12746
 Ophiura integra cf.      | OGL:Genomic:12747
 Ophiura integra cf.      | OGL:Genomic:12741
 Ophiura migrans          | OGL:Genomic:12793
 Ophiura migrans          | OGL:Genomic:12791
 Ophiura migrans          | OGL:Genomic:12792
 Ophiura migrans          | OGL:Genomic:12788
 Ophiura migrans          | OGL:Genomic:12801
 Ophiura migrans          | OGL:Genomic:12562
 Ophiura migrans          | OGL:Genomic:12794
 Ophiura migrans          | OGL:Genomic:12797
 Ophiura migrans          | OGL:Genomic:10445
 Ophiura migrans          | OGL:Genomic:10448
 Ophiura migrans          | OGL:Genomic:10446
 Ophiura migrans          | OGL:Genomic:12789
 Ophiura migrans          | OGL:Genomic:12798
 Ophiura migrans          | OGL:Genomic:12795
 Ophiura migrans          | OGL:Genomic:12802
 Ophiura migrans          | OGL:Genomic:12796
 Ophiura migrans          | OGL:Genomic:12803
 Ophiura migrans          | OGL:Genomic:12800
 Ophiura migrans          | OGL:Genomic:12790
 Ophiura migrans          | OGL:Genomic:12799
 Parapagurus rubra        | OGL:Genomic:16504
 Parapagurus rubra        | OGL:Genomic:16476
 Parapagurus rubra        | OGL:Genomic:16483
 Parapagurus rubra        | OGL:Genomic:16497
 Parapagurus rubra        | OGL:Genomic:16490
 Pinophyta                | OGL:Genomic:40693
 Pinophyta                | OGL:Genomic:40692
 Pterioida                | OGL:Genomic:3173
 Scinaia confusa          | OGL:Genomic:24440
 Sparlingia pertusa       | OGL:Genomic:24445
 Thalassia testudinum     | OGL:Genomic:34
 Thiosocius teredinicola  | OGL:Genomic:25849
 Turdus albicollis        | OGL:Genomic:10961
 Ulva lactuca             | OGL:Genomic:40701
 Ulva lactuca             | OGL:Genomic:40694
 Ulva lactuca             | OGL:Genomic:40698
 Ulva lactuca             | OGL:Genomic:40708
 Ulva lactuca             | OGL:Genomic:40699
 Ulva lactuca             | OGL:Genomic:40700
 Ulva lactuca             | OGL:Genomic:40695
 Ulva lactuca             | OGL:Genomic:40702
```

and that's possibly small enough for personal attention. (I did not know there were marine robins...)

**Comment by happiah-madson** [comment]
> feed that to search, then tools/request refresh

This search is the one on the main search page that will give me the the searched records, right? There are def taxa that we have that won't be WoRMS (looking at you bird robin) (side bar: there ARE sea robins, they're just fish: https://en.wikipedia.org/wiki/Triglidae AND SO COOL) but there are 8 that are in WoRMS. I think I can manually push those.

**Comment by dustymc** [comment]
>main search page 

Yup https://arctos.database.museum/search.cfm?guid_prefix=OGL%3AGenomic&taxon_name=Ophiura%20migrans then
<img width="216" alt="Screenshot 2025-01-08 at 08 34 54" src="https://github.com/user-attachments/assets/77047610-9cce-4431-8ce6-88afdde232dc" />

**Comment by happiah-madson** [comment]
Thank you! I think I have successfully done this. Hooray! I'm closing this issue...for now!
